### PR TITLE
Moved `customServer.url` to the request handler, now it is just `url`…

### PR DESCRIPTION
**Pull request by nmai**
…. Fixed getDir params as well (untested though).

Issue was with scope. No reason to keep a globally accessible reference to a URL (it gets changed with every request, causing conflicts).

### CODE REVIEW

**Comment by dekoding** [comment]
Very good. I hadn't considered a scope issue, but of course, it's obvious in retrospect. Any asynchronous function that changes the value of the global reference could easily yank that value out from under another asynchronous function.

**Comment by nmai** [comment]
@GoodDamon Haha, everything is obvious in retrospect. I just submitted the answer on SO with a broader explanation (sorry for the slow turnaround, I tend to write, delete, and re-write a lot!) but it sounds like you got this figured out now.

Also, awesome that you're taking the time to start from the bottom and build a fundamental understanding of how these tools work. :+1:  When I first started working with Node, I would just `npm install` a bunch of frameworks and that would be it. Needless to say, this approach did not work for very long!

**Comment by dekoding** [comment]
Thanks! Yeah, I've had to fight the urge to just install packages that make things too easy. I have a few different projects I want to create, and if I don't actually know my way around Node before I start trying to create them, I'm going to make huge mistakes, code myself into corners, and just plain fail. Better to get the hard part out of the way first and _actually understand_ what I'm doing, so when I start building the projects I really care about, I don't screw it up too badly. :)

**Comment by nmai** [comment]
Yup, and once you understand how this all works, you'll find that plenty of widely-used modules are actually quite simple in terms of implementation. For example, Express and all of its middleware is essentially just a linear chain of callbacks attached to node's HTTP server. Based off what you have now, you could create a similar framework.

_As a (unrelated) side note_\- while Express' implementation is beautifully simple and node-like, it does impose certain constraints. For example, if you wanted to hit a certain function in your chain of middleware more than once, pretty much the only way to accomplish this is by sending another HTTP request. I would say this is not an issue for most devs, but for those that need something more flexible they might use something like Hapi.js instead (hapi was actually built by a dev who gave up trying to shoehorn Express into a role it was never built for). 

Basically, if you decide to pick up a new tool/lib/framework, just be aware of what it is and what it isn't. By learning the basics of node and trying stuff out for yourself, you're going to be able to make more informed decisions when it comes to these things. I think you're headed in the right direction :)

**Comment by dekoding** [comment]
Thank you! I'm impressed by Express (and Hapi.js), but I don't think I'll be using either of them any time soon for my own projects.
